packages/mailer/src/Testing/TestingMailer.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Tempest\Mail\Testing;
+
+use Tempest\EventBus\EventBus;
+use Tempest\Mail\Email;
+use Tempest\Mail\EmailSent;
+use Tempest\Mail\EmailToSymfonyEmailMapper;
+use Tempest\Mail\Mailer;
+use Tempest\Mail\MailerConfig;
+use Tempest\View\ViewRenderer;
+
+use function Tempest\map;
+
+final class TestingMailer implements Mailer
+{
+    public function __construct(
+        private readonly MailerConfig $mailerConfig,
+        private readonly ViewRenderer $viewRenderer,
+        private readonly ?EventBus $eventBus = null,
+    ) {}
+
+    /**
+     * List of emails that would have been sent.
+     *
+     * @var array<Email>
+     */
+    private(set) array $sent = [];
+
+    public function send(Email $email): SentTestingEmail
+    {
+        $this->sent[] = $email;
+
+        $symfonyEmail = map($email)
+            ->with(fn (Email $from) => new EmailToSymfonyEmailMapper($this->mailerConfig, $this->viewRenderer)->map($from, null))
+            ->do();
+
+        $this->eventBus?->dispatch(new EmailSent($email));
```

Comment:
I'm going to find it confusing that we have `EmailSent` and `SentEmail`, and both meaning something different. Can't we merge both of them in the `EmailSent` event? I noticed there's a testing implementation for `SentEmail`, but I figure those assertion methods can be moved to somewhere else.